pkg/didcomm/protocol/introduce/service.go

```
@@ -248,22 +257,30 @@ func (s *Service) doHandle(msg *service.DIDCommMsg, outbound bool) (*metaData, e
 
 // InvitationReceived is used to finish the state machine
 // the function should be called by didexchange after receiving an invitation
-func (s *Service) InvitationReceived(thID string) error {
+func (s *Service) InvitationReceived(msg service.StateMsg) error {
+	if msg.StateID != "invited" || msg.Type != service.PostState {
+		return nil
+	}
+
+	if msg.Msg.Header == nil || msg.Msg.Header.Thread.PID == "" {
+		return nil
+	}
+
 	payload, err := json.Marshal(&model.Ack{
 		Type:   AckMsgType,
 		ID:     uuid.New().String(),
-		Thread: &decorator.Thread{ID: thID},
+		Thread: &decorator.Thread{ID: msg.Msg.Header.Thread.PID},
 	})
 	if err != nil {
 		return fmt.Errorf("invitation received marshal: %w", err)
 	}
 
-	msg, err := service.NewDIDCommMsg(payload)
+	didMsg, err := service.NewDIDCommMsg(payload)
 	if err != nil {
 		return fmt.Errorf("invitation received new DIDComm msg: %w", err)
 	}
 
-	_, err = s.HandleInbound(msg)
+	_, err = s.HandleInbound(didMsg)
```

i assume the service will silently ignore unrelated threads?